Automattic/woocommerce-payments
# Fix connection URL when user is already onboarded

Fixes #3945 

#### Changes proposed in this Pull Request

On WooCommerce Admin page, the "Set up payments" option sometimes is not completed/checked. When the user clicks on it, the user will be redirected to the onboarding page. If the user is already onboarded, the error will be shown that user cannot access this page. The user shouldn't access this page if it is already onboarded, but the real issue is the redirection URL. If the user is already onboarded, but payments setup is not yet finished, the user should be redirected to the set-up page instead of connect (onboarding) page.





#### Testing instructions

1. Create new WooCommerce Payments account (you can use https://jurassic.ninja/ with WCPay devs plugin)
2. Onboard new account using test data.
3. Make sure that account is restricted. To make account restricted, you can finish onboarding process and remove some user data (like address or something similar) on Stripe connect webpage
4. Go to WCPay Dev plugin section and delete cache data to avoid possible issues
5. Make sure that your account status is restricted because code that is used to determinate does account needs setup checks for errors
6. Go to WooCommerce admin page `wp-admin/admin.php?page=wc-admin` and click on "Set up payments" option
7. Click on Finish setup button
8. A new screen should be opened where a button will be shown with text 'Set up' instead of 'Onboard'
9. When clicked on that button, the page will be redirected to the WooCommerce Payments configuration page where you can add multiple payments methods, ...

*

-------------------


**Post merge**
